enclave/CMakeLists.txt

```diff
@@ -32,61 +32,72 @@ elseif (OE_TRUSTZONE)
   message("TODO: ADD ARM files.")
 endif ()
 
-add_enclave_library(
-  oeenclave
-  STATIC
-  attest_plugin.c
-  ../common/attest_plugin.c
-  ../common/custom_claims.c
-  ../common/datetime.c
-  ../common/sha.c
-  asym_keys.c
-  link.c
-  tls_cert.c
-  ${PLATFORM_SRC})
+macro (add_library_oeenclave_wrapper lib_name)
 
-maybe_build_using_clangw(oeenclave)
+  add_enclave_library(
+    ${lib_name}
+    STATIC
+    attest_plugin.c
+    ../common/attest_plugin.c
+    ../common/custom_claims.c
+    ../common/datetime.c
+    ../common/sha.c
+    asym_keys.c
+    link.c
+    tls_cert.c
+    ${PLATFORM_SRC})
 
-if (CMAKE_C_COMPILER_ID MATCHES GNU)
-  enclave_compile_options(oeenclave PRIVATE -Wjump-misses-init)
+  maybe_build_using_clangw(${lib_name})
 
-  if (OE_TRUSTZONE)
-    target_compile_options(oeenclave PUBLIC ${OE_TZ_TA_C_FLAGS})
+  if (CMAKE_C_COMPILER_ID MATCHES GNU)
+    enclave_compile_options(${lib_name} PRIVATE -Wjump-misses-init)
+
+    if (OE_TRUSTZONE)
+      target_compile_options(${lib_name} PUBLIC ${OE_TZ_TA_C_FLAGS})
+    endif ()
   endif ()
-endif ()
 
-enclave_enable_code_coverage(oeenclave)
+  enclave_enable_code_coverage(${lib_name})
 
-# Add location of the oeedger8r-generated trusted headers for internal
-# functions implemented via EDL files.
-enclave_include_directories(oeenclave PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/core)
+  # Add location of the oeedger8r-generated trusted headers for internal
+  # functions implemented via EDL files.
+  enclave_include_directories(${lib_name} PRIVATE
+                              ${CMAKE_CURRENT_BINARY_DIR}/core)
 
-enclave_link_libraries(oeenclave PUBLIC oelibc)
+  enclave_link_libraries(${lib_name} PUBLIC oelibc)
 
-if (OE_TRUSTZONE)
-  enclave_link_libraries(oeenclave PUBLIC oelibutee)
-endif ()
+  if (OE_TRUSTZONE)
+    enclave_link_libraries(${lib_name} PUBLIC oelibutee)
+  endif ()
 
-set_enclave_property(TARGET oeenclave PROPERTY ARCHIVE_OUTPUT_DIRECTORY
-                     ${OE_LIBDIR}/openenclave/enclave)
-install_enclaves(
-  TARGETS
-  oeenclave
-  EXPORT
-  openenclave-targets
-  ARCHIVE
-  DESTINATION
-  ${CMAKE_INSTALL_LIBDIR}/openenclave/enclave)
+  set_enclave_property(TARGET ${lib_name} PROPERTY ARCHIVE_OUTPUT_DIRECTORY
+                       ${OE_LIBDIR}/openenclave/enclave)
+  install_enclaves(
```

Comment:
We need to also make this configurable so that we don't publish the version with loophole.